### PR TITLE
Improve CI checks

### PR DESCRIPTION
**Pull request by maelanleborgne**
- Add phpstan lvl 8 to CI
- Fix code to pass phpstan check
- Upgrade phpunit to 10.5 + get rid of symfony/phpunit-bridge
- Add php 8.3 to the list of tested versions

### CODE REVIEW

File: src/SensiolabsTypeScriptBundle.php [hunk 1]
```diff
@@ -8,8 +8,12 @@
 
 class SensiolabsTypeScriptBundle extends Bundle
 {
-    public function getContainerExtension(): ?ExtensionInterface
+    public function getContainerExtension(): ExtensionInterface
     {
-        return $this->extension ?? $this->extension = new SensiolabsTypeScriptExtension();
+        if (null === $this->extension || false === $this->extension) {
+            $this->extension = new SensiolabsTypeScriptExtension();
+        }
```
Comment (Jean-Beru) [inline src/SensiolabsTypeScriptBundle.php:15]: `?:` will produce the same result, right?

File: src/AssetMapper/TypeScriptPublicPathAssetPathResolver.php [hunk 2]
```diff
@@ -23,6 +23,9 @@ public function resolvePublicPath(string $logicalPath): string
 
     public function getPublicFilesystemPath(): string
     {
+        if (!method_exists($this->decorator, 'getPublicFilesystemPath')) {
```
Comment (Jean-Beru) [inline src/AssetMapper/TypeScriptPublicPathAssetPathResolver.php:26]: What about using an Interface instead to make it more maintenable?

Comment (maelanleborgne) [reply src/AssetMapper/TypeScriptPublicPathAssetPathResolver.php:26]: This method was part of the interface in 6.3 but removed in 6.4 ([see this commit](https://github.com/symfony/symfony/commit/48a2d689e6e91c943d931d30e90555726ac6f7c7#diff-8493cb0e782a4fe3c754664c4fa0101a3fcf8bd8bdc504b48d614bf3849f1b3c)). This check is for backward compatibility

File: tests/fixtures/other_dir/file_in_other_dir
[the remark this answers is not in the document]
Comment (ktherage) [reply tests/fixtures/other_dir/file_in_other_dir:1]: LGTM then.

**Pull request merged**